resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -257,6 +304,11 @@ public void operationComplete(ChannelFuture future) throws Exception {
         });
     }
 
+    // Only here to override in unit tests.
+    int dnsRedirectPort(@SuppressWarnings("unused") InetAddress server) {
```

Comment:
SonarQube doesn't seem to understand the annotation unfortunately. We may want to upgrade it to a newer version to see whether this problem is gone. There was a problem with MariaDB last time I tried to upgrade though.